internal/security/secretstore/kongadminapi.go

```diff
@@ -0,0 +1,241 @@
+type KongAdminAPI struct {
+	secrets  KongAdminAPISecrets
+	prefixes KongAdminAPIPrefixes
+	paths    KongAdminAPIPaths
+}
+
+// KongAdminAPIPrefixes is used for prettier & scoped logging messages
+type KongAdminAPIPrefixes struct {
+	cmdText  string
+	errText  string
+	infoText string
+}
+
+// KongAdminAPIKey is the struct that holds all the secrets
+type KongAdminAPISecrets struct {
+	public  ECKey
+	private ECKey
+	jwt     JWT
+}
+
+// KongAdmingAPIPaths are mapped to the paths defined in configuration.toml
+type KongAdminAPIPaths struct {
+	template string
+	config   string
+	jwt      string
+}
+
+// ECKey is a simple struct to hold public/private keys in different formats
+type ECKey struct {
+	encoded []byte
+	pem     string
+}
+
+// JWT is a simple struct to hold the necessary JWT string values
+type JWT struct {
+	issuer string
+	signed string
+}
+
+// NewKongAdminAPI is a generic constructor to return an instance of KongAdminAPI
+func NewKongAdminAPI() KongAdminAPI {
+	return KongAdminAPI{}
+}
+
+// Setup creates a Kong declarative configuration file and a JWT.
+//
+// The configuration file is a Kong declarative configuration file in YML format
+// that is written to an area where the Kong service has access and can import
+// the file.
+//
+// The JWT is saved to a JSON based file
+func (k *KongAdminAPI) Setup() error {
+
+	// Verify prefixes and paths
+	err := k.verifyPrefixes()
+	if err != nil {
+		return err
+	}
+
+	err = k.verifyPaths()
+	if err != nil {
+		return err
+	}
+
+	// Generate the keys for the Kong Admin API
+	key, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		return fmt.Errorf("%s Failed to generate private/public key pair: %w", k.prefixes.errText, err)
+	}
+
+	// Get an encoded version of the keys
+	k.secrets.private.encoded, _ = x509.MarshalECPrivateKey(key)
+	k.secrets.public.encoded, _ = x509.MarshalPKIXPublicKey(&key.PublicKey)
+
+	// PEM encode the keys
+	k.secrets.private.pem = string(pem.EncodeToMemory(&pem.Block{Type: "PRIVATE KEY", Bytes: k.secrets.private.encoded}))
+	k.secrets.public.pem = string(pem.EncodeToMemory(&pem.Block{Type: "PUBLIC KEY", Bytes: k.secrets.public.encoded}))
+
+	// When complete, clear the secrets portion of the KongAdminAPI struct
+	defer k.clearSecrets()
+
+	// Read in the configuration template
+	configTemplateBytes, err := ioutil.ReadFile(k.paths.template)
+	if err != nil {
+		return fmt.Errorf("%s Failed to read config template from file %s: %w", k.prefixes.errText, k.paths.template, err)
+	}
+
+	// Set random string for JWT issuer payload value
+	k.secrets.jwt.issuer = helper.GenerateRandomString(32)
+
+	// Insert public key
+	configTemplateText := strings.Replace(string(configTemplateBytes),
+		"<<INSERT-ADMIN-PUBLIC-KEY>>", strings.ReplaceAll(strings.TrimSpace(k.secrets.public.pem), "\n", "\n    "), -1)
+
+	// Insert issuer
+	configTemplateText = strings.Replace(configTemplateText,
+		"<<INSERT-ADMIN-JWT-ISSUER-KEY>>", k.secrets.jwt.issuer, -1)
+
```

Comment:
I feel like it's a hassle to implement all of that (maybe including additional dependencies?) for 2 simple values that just need to be swapped out. And since this is a YML file, I have to handle the special snowflake-esque spacing on the public key insertion or the Kong config parser will complain.